### ISSUE TITLE
Submission: group_13: wine-origin-prediction

### ISSUE THREAD

**Issue by juliaeveritt** [UBC-MDS/data-analysis-review-2023#6]
**Submitting authors:**  @sean-m-mckay @hbandukw @yimengxia @juliaeveritt

**Repository:** https://github.com/UBC-MDS/wine-origin-prediction/tree/main
**Report link:** https://ubc-mds.github.io/wine-origin-prediction/wine_classification_report.html
**Abstract/executive summary:** https://github.com/UBC-MDS/wine-origin-prediction/blob/main/README.md

**Editor:** @ttimbers 
**Reviewer:** <REVIEWER1> <REVIEWER2> <REVIEWER3>

- [ ] I agree to abide by [MDS's Code of Conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by JohnShiuMK** [comment]
## Data analysis review checklist

### Reviewer: JohnShiuMK

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [ ] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [ ] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1 hour

### Review Comments: 
- It is better to save the raw data explicitly into /data/raw/, after `from ucimlrepo import fetch_ucirepo` and `data = fetch_ucirepo(id=109)`, so that we are able to read the raw data without running the codes
- The report would look nicer if you could hide the codes at the beginning
- It would be a big plus if an example is provided at the end of the report, to illustrate some predictions made by the Logistic Regression, and some discussions to reinforce how the predictions benefit the industry.
- Other than these, the report and codes are well written, tests are comprehensive and the README is very clear.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by tangyl92** [comment]
## Data analysis review checklist

### Reviewer: <GITHUB_USERNAME> 

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 

### Review Comments: 
Strength:
- The project effectively demonstrates the use of logistic regression in classifying wines based on their physiochemical properties. 
- The introduction provides a strong context for the importance of wine identification in the global market and sets up the necessity for an efficient, data-driven approach.
- You have clearly outlined the source and composition of your dataset, which enhances the transparency and replicability of the research.
- The use of a grid search for hyperparameter optimization and cross-validation are good practices that lend credibility to your model
- The paper is well-structured and clearly written, making it accessible to readers with varying levels of expertise in machine learning.

Suggestions:
- It would be useful to include more detailed results from the model, such as confusion matrices or ROC curves, to better understand its performance. 
- It would be immensely beneficial if you could provide the coefficient values and their corresponding interpretations from your model.


#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by dorisyycai** [comment]
## Data analysis review checklist

### Reviewer: @dorisyycai

### Conflict of interest

- [X] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [X] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [X] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [X] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [X] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [X] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [X] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [X] **Style guidelides:** Does the code adhere to well known language style guides?
- [X] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [X] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [X] **Data:** Is the raw data archived somewhere? Is it accessible?
- [X] **Computational methods:** Is all the source code required for the data analysis available?
- [X] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [X] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [X] **Authors:** Does the report include a list of authors with their affiliations?
- [X] **What is the question:** Do the authors clearly state the research question being asked?
- [X] **Importance:** Do the authors clearly state the importance for this research question?
- [X] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [X] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [X] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [X] **Conclusions:** Are the conclusions presented by the authors correct? 
- [X] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [X] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1.5h

### Review Comments: 

Overall it is an awesome project. The model built by the team has an excellent performance of accuracy score of 0.96 and a F1 score of 0.96. Below is something I think it would be nice to include:

- Dependencies section on README clearly listed dependencies required for running the project within the local environment using environment.yml. As there is another option, the docker container, created to run this project, it would be better to include the dependencies on dockerfile in this section. 

- Main report clearly includes all references related to the project and figures are properly generated on the report. It would be nice if the report can also include captions for each figure, as well as the figure number for reference within the analysis, which would be easier for users to follow the analysis. Also, hiding the code cells on the main report may reduce distractions from the main analysis that the authors want to deliver.

- Looks like the project includes hyperparameter tuning as it includes a line chart of accuracy vs hyperparameter C for training set and validation set. May be more comments on what value of hyperparameter C is chosen for the best model would help the audience better understand the model chosen for this project.

- I wasn’t able to reproduce the report in the container. I successfully opened up the container using `docker compose up` of setup option 1. However, when I tried to run multiple scripts listed in the data analysis section and tests using pytest, I encountered an error of a module not found. I wonder if the dockerfile copies dependencies properly from the environment.yml file or my user mistakes. I may need some help from the project team to guide me how to run these scripts properly.

![Untitled copy](https://github.com/UBC-MDS/data-analysis-review-2023/assets/100008826/172cf78a-e2b2-4938-a6d6-4fafd4f324ad)


Overall it is a great project!

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.

**Comment by LilyTao0531** [comment]
## Data analysis review checklist

### Reviewer: @LilyTao0531

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [x] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [x] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing: 1.5

### Review Comments: 

Generally, it is great idea to use machine learning to provide insights for problems that require high level of expertise.  The software setup process is smoothe without error, and I really enjoyed reading the report. Below is the specific things I would like to mention:

-  I think the illustrations are quite intuitive in terms of revealing the distributions of the features. Formatting all features in one page facilitates the comparison across different features. From my perspective, it would be better if the figures can be on a larger scale so that the x-axis and y-axis are more clear, provided that they vary from feature to feature.

- The code chunks give me a general idea about the packages that are necessary in this report, but I think hiding them can better highlights the main idea without interrupting the flow of the report.

- It seems to me that the report does not mention data processing explicitly, so I assume that the data is quite ready upon download. I guess it would be better if there is more information about data quality, for example, if there is any missing value, and the statistics of the features like mean and standard deviation.

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.
